onedal/backend/tests/test_data.py

```diff
@@ -0,0 +1,127 @@
+#===============================================================================
+# Copyright 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+import pytest
+import numpy as np
+from numpy.testing import assert_allclose
+from sklearn import datasets, metrics
+from onedal.prims import linear_kernel
+
+from sklearn.utils.estimator_checks import check_estimator
+import sklearn.utils.estimator_checks
+
+
+def _test_input_format_c_contiguous_numpy(dtype):
+    rng = np.random.RandomState(0)
+    x_default = np.array(5 * rng.random_sample((10, 4)), dtype=dtype)
+
+    x_numpy = np.asanyarray(x_default, dtype=dtype, order='C')
+    assert x_numpy.flags.c_contiguous
+    assert not x_numpy.flags.f_contiguous
+    assert not x_numpy.flags.fnc
+
+    expected = linear_kernel(x_default)
+    result = linear_kernel(x_numpy)
+    assert_allclose(expected, result)

```

Comment:
Why this file named like it contains tests on data module when actually it is testing how linear kernel works?

Reply:
Linear kernels are the easiest algorithm. All data manipulation is done inside each algorithm. There is a check for different formats, and different cases of data

Reply:
But it is the check no on the data module but on the functionality of the linear kernel. The name of the file says it is testing the data module. Please, put these tests closer to linear kernel functionality since it is testing exactly this class